mithril-doc/src/markdown_formatter.rs

```diff
@@ -0,0 +1,439 @@
+                let align_left = header.chars().next().map(|c| c == ':').unwrap_or(false);
+                let align_right = header.chars().last().map(|c| c == ':').unwrap_or(false);
+                let label = &header[(if align_left { 1 } else { 0 })
+                    ..(header.len() - (if align_right { 1 } else { 0 }))];
+                (label, align_left, align_right)
+            })
+            .collect::<Vec<(&str, bool, bool)>>();
+
+        let sublines = headers
+            .iter()
+            .map(|(label, left, right)| {
+                format!(
+                    "{}{}{}",
+                    if *left { ":" } else { "-" },
+                    "-".repeat(label.len()),
+                    if *right { ":" } else { "-" }
+                )
+            })
+            .collect::<Vec<String>>();
+
+        let labels = headers
+            .iter()
+            .map(|(label, _, _)| label.to_string())
+            .collect::<Vec<String>>();
+
+        format!("| {} |\n|{}|", labels.join(" | "), sublines.join("|"))
+    }
+}
+
+pub fn doc_markdown_with_config(cmd: &mut Command, struct_doc: Option<&StructDoc>) -> String {
+    // See: https://github1s.com/clap-rs/clap/blob/HEAD/clap_builder/src/builder/command.rs#L1989
+
+    fn format_parameters(cmd: &Command, struct_doc: Option<&StructDoc>) -> String {
+        if cmd
+            .get_arguments()
+            .filter(|arg| argument_to_document(arg))
+            .peekable()
+            .peek()
+            .is_some()
+        {
+            let mut command_parameters = extract_clap_info::extract_parameters(cmd);
+            if let Some(config_doc) = struct_doc {
+                if !config_doc.data.is_empty() {
+                    command_parameters = command_parameters.merge_struct_doc(config_doc);
+                }
+            }
+
+            let parameters_table = doc_config_to_markdown(&command_parameters);
+
+            let parameters_explanation = "\n\
+                The configuration parameters can be set in either of the following ways:\n\
+                \n\
+                1. In a configuration file, depending on the `--run-mode` parameter. If the runtime mode is `testnet`, the file is located in `./conf/testnet.json`.\n\
+                \n\
+                2. The value can be overridden by an environment variable with the parameter name in uppercase.\n\
+                ";
+            format!("{}\n{}", parameters_explanation, parameters_table)
+        } else {
+            "".to_string()
+        }
+    }
+
+    fn format_subcommand(cmd: &Command) -> String {
+        let sub_commands = &mut cmd.get_subcommands().peekable();
+        if sub_commands.peek().is_some() {
+            let subcommands_lines = sub_commands
+                .map(|command| {
+                    vec![
+                        format!("**{}**", command.get_name()),
+                        command.get_all_aliases().collect::<Vec<&str>>().join(","),
```

Comment:
Can we put a `-` instead of an empty string in the aliases list?

Reply:
I'll remove subcommand aliases because they do not appear in the current documentation.